### PR TITLE
MagenticOne Orchestrator Fixes

### LINKED ISSUE #4177: AgentChat port of MagenticOneOrchestrator needs to validate json
AgentChat port of MagenticOneOrchestrator needs to validate json here:

https://github.com/microsoft/autogen/blob/a27a130b4553c9002797e82a29bbc6db90b1bc92/python/packages/autogen-agentchat/src/autogen_agentchat/teams/_group_chat/_magentic_one/_magentic_one_orchestrator.py#L214C4-L215C55

The original version has code for this:

https://github.com/microsoft/autogen/blob/a27a130b4553c9002797e82a29bbc6db90b1bc92/python/packages/autogen-magentic-one/src/autogen_magentic_one/agents/orchestrator.py#L215-L260

### PR DESCRIPTION
**Pull request by husseinmozannar**
## Why are these changes needed?

Add back some verification logic to MagenticOne Orchestrator.

Base the orchestrator on the base group chat manager.

Remaining:

- add intermediate logging
- update tests

## Related issue number

Will resolve #4177 

#3756  - Updates to this PR

### CODE REVIEW

**Comment by husseinmozannar** [comment]
@ekzhu still working on this, but curious, is this better for integrating the M1 orchestrator as an actual group chat manager? Anything else that would improve it?

**Review by ekzhu — APPROVED** [review]
Looks good. Don't forget about adding unit tests for the magentic one group chat. Use ReplayChatCompletionClient #4412

**Comment by ekzhu** [comment]
> @ekzhu still working on this, but curious, is this better for integrating the M1 orchestrator as an actual group chat manager? Anything else that would improve it?

Yes it is good to make it a group chat manager. It allows us to refactor group chat manager logic into a developer facing API.

**Comment by ekzhu** [comment]
Also, remember to add API docs with examples and argument list for MagenticOneGroupChat

#4413

**Comment by husseinmozannar** [comment]
Thanks Eric!

Added two simple tests and API docs, and ironed a few bugs + trace logger.

I ran into a mypy error because of overriding handle_agent_response and handle_start, added # type ignore because I couldn't figure out how to fix.

**Comment by ekzhu** [comment]
One last small lint error see checks.

**Comment by ekzhu** [comment]
> Thanks Eric!
> 
> 
> 
> Added two simple tests and API docs, and ironed a few bugs + trace logger.
> 
> 
> 
> I ran into a mypy error because of overriding handle_agent_response and handle_start, added # type ignore because I couldn't figure out how to fix.
> 
> 
> 
> Please merge when appropriate

Overriding handlers causes mypy error is a know issue.

File: python/packages/autogen-agentchat/src/autogen_agentchat/teams/_group_chat/_magentic_one/_magentic_one_group_chat.py [hunk 1]
```diff
@@ -13,18 +13,62 @@
...
+
+    MagenticOneGroupChat with one assistant agent:
+
+        .. code-block:: python
+
+            import asyncio
+            from autogen_ext.models import OpenAIChatCompletionClient
+            from autogen_agentchat.agents import AssistantAgent
+            from autogen_agentchat.teams import MagenticOneGroupChat
+            from autogen_agentchat.task import TextMentionTermination, Console
```
Comment (ekzhu) [inline python/packages/autogen-agentchat/src/autogen_agentchat/teams/_group_chat/_magentic_one/_magentic_one_group_chat.py:42]:
```suggestion
            from autogen_agentchat.task import Console
```

**Pull request merged**